docs/src/main/asciidoc/security-authentication-mechanisms.adoc

```diff
@@ -352,13 +352,82 @@ quarkus.http.auth.policy.role-policy-cert.roles-allowed=user,admin     <2>
 
 Given the preceding configuration, the request is authorized if the client certificate's CN attribute is equal to `alice` or `bob` and forbidden if it is equal to `jdoe`.
 
+[[use-cert-attrs-to-augment-identity]]
 ==== Using certificate attributes to augment SecurityIdentity
 
 You can always register `SecurityIdentityAugmentor` if the automatic <<map-certificate-attributes-to-roles>> option does not suit.
 Custom `SecurityIdentityAugmentor` can check the values of different client certificate attributes and augment the `SecurityIdentity` accordingly.
 
 For more information about customizing `SecurityIdentity`, see the xref:security-customization.adoc#security-identity-customization[Security identity customization] section in the Quarkus "Security tips and tricks" guide.
 
+[[mtls-programmatic-set-up]]
+==== Set up the mutual TLS client authentication programmatically
+
+The `io.quarkus.vertx.http.security.HttpSecurity` CDI event allows to configure the mutual TLS client authentication mechanism programmatically like in the example below:
+
+[source,java]
+----
+package org.acme.http.security;
+
+import io.quarkus.vertx.http.security.HttpSecurity;
+import jakarta.enterprise.event.Observes;
+
+public class MutualTlsClientAuthConfig {
+
+    void configure(@Observes HttpSecurity httpSecurity) {
+        httpSecurity.mTLS(); <1>
+    }
+
+}
+----
+<1> Enable the mutual TLS client authentication.
+
+The above-mentioned configuration has a similar effect as following configuration in the 'application.properties' file:
+
+[source,properties]
+----
+quarkus.http.ssl.client-auth=REQUIRED
+----
+
+This API also enables you to use certificate attributes to augment `SecurityIdentity`:
+
+[source,java]
+----
+package org.acme.http.security;
+
+import io.quarkus.vertx.http.security.HttpSecurity;
+import io.quarkus.vertx.http.security.MTLS;
+import jakarta.enterprise.event.Observes;
+
+public class MutualTlsClientAuthConfig {
+
+    void configure(@Observes HttpSecurity httpSecurity) {
+            var builder = MTLS.builder();
```

Comment:
IMHO we just show only how you can map roles with certificates, to keep focus on the new MTLS option

Reply:
> IMHO we just show only how you can map roles with certificates, to keep focus on the new MTLS option

never mind, I took a guess and I think I got it